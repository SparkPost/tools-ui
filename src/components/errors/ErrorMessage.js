import React, { Component } from 'react';
import Icon from 'components/Icon';

import './ErrorMessage.scss';

/**
 * Produces an error message block
 */
class ErrorMessage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }

  toggleDetails() {
    this.setState({
      open: !this.state.open
    });
  }

  renderToggleButton() {
    const { open } = this.state;
    return (
      <a className='errorMessage__link' onClick={() => this.toggleDetails()}>{open ? 'Hide Details' : 'View Details'}</a>
    );
  }

  render() {
    const { icon, friendly, details } = this.props;
    const { open } = this.state;

    return (
      <div className='errorMessage'>
        <div className='errorMessage__friendly'>
          {icon && <Icon name={icon} />}
          {friendly}
          {details && this.renderToggleButton()}
        </div>

        {(details && open) &&
          <div className='errorMessage__details'>
            {details}
          </div>}
      </div>
    );
  }
}

ErrorMessage.defaultProps = {
  friendly: 'Something happened!',
  details: null,
  icon: 'exclamation-circle'
};

ErrorMessage.propTypes = {
  friendly: React.PropTypes.string,
  details: React.PropTypes.string,
  icon: React.PropTypes.string
};

export default ErrorMessage;
