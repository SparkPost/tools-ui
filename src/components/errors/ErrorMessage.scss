@import "~@sparkpost/styles/src/assets/scss/shared/config";
@import "~@sparkpost/styles/src/assets/scss/shared/functions";

.errorMessage {
  display: block;
  margin-bottom: 1rem;
  padding: rem(9) rem(15);

  background: grayscale(white);
  box-shadow: shadow(1);

  font-weight: 400;
  font-size: rem(13);
  color: color(alert);
  text-align: left;

  .fa {
    display: inline-block;
    margin-right: rem(15);
    font-size: rem(15);
  }
}
