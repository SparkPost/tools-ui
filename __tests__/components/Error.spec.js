import React from 'react';
import renderer from 'react-test-renderer';

import ErrorMessage from 'components/errors/ErrorMessage';

describe('ErrorMessage component', () => {

  test('should render the default message with no props', () => {
    expect(renderer.create(<ErrorMessage />)).toMatchSnapshot();
  });

  test('should render the a supplied icon', () => {
    expect(renderer.create(<ErrorMessage icon='fa-bolt' />)).toMatchSnapshot();
  });

  test('should render correctly with an error message', () => {
    expect(renderer.create(<ErrorMessage friendly='some string message' />)).toMatchSnapshot();
  });

  test('should render correctly with an error message with details', () => {
    expect(renderer.create(<ErrorMessage friendly='some string message' details='extra details' />)).toMatchSnapshot();
  });

});
