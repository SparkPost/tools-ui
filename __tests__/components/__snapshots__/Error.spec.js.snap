exports[`ErrorMessage component should render correctly with a supplied icon 1`] = `
<div
  className="errorMessage">
  <div
    className="errorMessage__text">
    <i
      className="fa fa-fa-bolt "
      style={
        Object {
          "transition": "all 200ms ease-out",
        }
      } />
    <p
      className="errorMessage__friendly">
      Something happened!
    </p>
  </div>
</div>
`;

exports[`ErrorMessage component should render correctly with an error message 1`] = `
<div
  className="errorMessage">
  <div
    className="errorMessage__text">
    <i
      className="fa fa-exclamation-circle "
      style={
        Object {
          "transition": "all 200ms ease-out",
        }
      } />
    <p
      className="errorMessage__friendly">
      some string message
    </p>
  </div>
</div>
`;

exports[`ErrorMessage component should render correctly with an error message and details 1`] = `
<div
  className="errorMessage">
  <div
    className="errorMessage__text">
    <i
      className="fa fa-exclamation-circle "
      style={
        Object {
          "transition": "all 200ms ease-out",
        }
      } />
    <a
      className="errorMessage__link"
      onClick={[Function]}>
      View Details
    </a>
    <p
      className="errorMessage__friendly">
      some string message
    </p>
  </div>
</div>
`;

exports[`ErrorMessage component should render the default message with no props 1`] = `
<div
  className="errorMessage">
  <div
    className="errorMessage__text">
    <i
      className="fa fa-exclamation-circle "
      style={
        Object {
          "transition": "all 200ms ease-out",
        }
      } />
    <p
      className="errorMessage__friendly">
      Something happened!
    </p>
  </div>
</div>
`;
